tests/template_tests/test_loaders.py

```diff
@@ -238,6 +244,27 @@ def test_case_sensitivity(self):
             check_sources('index.html', ['/dir1/index.html', '/DIR2/index.html'])
             check_sources('/DIR1/index.HTML', ['/DIR1/index.HTML'])
 
+    def test_file_does_not_exist(self):
+        with self.assertRaises(TemplateDoesNotExist):
+            self.engine.get_template('doesnotexist.html')
+
+    @unittest.skipIf(
+        sys.platform == 'win32',
+        "Python on Windows doesn't have working os.chmod().",
+    )
+    def test_permissions_error(self):
+        with tempfile.NamedTemporaryFile() as tmpfile:
+            tmpdir = os.path.dirname(tmpfile.name)
+            tmppath = os.path.join(tmpdir, tmpfile.name)
+            os.chmod(tmppath, 0o0222)
+            with self.set_dirs([tmpdir]):
+                with self.assertRaisesMessage(IOError, 'Permission denied'):
+                    self.engine.get_template(tmpfile.name)
+
+    def test_notafile_error(self):
```

Comment:
Does the raised exception have the same `errno` on both platforms?

If that's the case you could do:

```python
with self.assertRaises(IOError) as ctx:
    # ....
self.assertEqual(ctx.exception.errno, errno.WHATEVER)
```